airflow-core/docs/howto/deadline-alerts.rst

```diff
@@ -0,0 +1,312 @@
+            interval=timedelta(minutes=30),
+            callback=AsyncCallback(
+                SlackNotifier,
+                kwargs={
+                    "slack_conn_id": "slack_default",
+                    "channel": "#alerts",
+                    "text": "DAG 'slack_deadline_alert' still running after 30 minutes.",
+                    "username": "Airflow Alerts",
+                },
+            ),
+        ),
+    ):
+        EmptyOperator(task_id="example_task")
+
+Creating Custom Callbacks
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can create custom callbacks for more complex handling. The ``callback_kwargs`` specified in
+the ``DeadlineAlert`` are passed to the callback function, if any are provided.  **Synchronous callbacks**
+(standard python methods) can be defined in the dag bundle and are run in the Executor.  **Asynchronous
+callbacks** must be defined somewhere in the Triggerer's system path.
+
+.. note::
+    Regarding Async Custom Deadline callbacks:
+
+    * Async callbacks are executed by the Triggerer, so users must ensure they are importable by the Triggerer.
+    * One easy way to do this is to place the callback as a top-level method in a new file in the plugins folder.
+    * The Triggerer will need to be restarted when a callback is added or changed in order to reload the file.
+
+A **custom synchronous callback** might look like this:
+
+.. code-block:: python
+
+    from datetime import timedelta
+
+    from airflow import DAG
+    from airflow.providers.standard.operators.empty import EmptyOperator
+    from airflow.sdk.definitions.deadline import DeadlineAlert, DeadlineReference, SyncCallback
+
+
+    def custom_synchronous_callback(**kwargs):
+        """Handle deadline violation with custom logic."""
+        print(f"Deadline exceeded for DAG {kwargs.get("dag_id")}!")
+        print(f"Alert type: {kwargs.get("alert_type")}")
+        # Additional custom handling here
+
+
+    with DAG(
+        dag_id="custom_deadline_alert",
+        deadline=DeadlineAlert(
+            reference=DeadlineReference.DAGRUN_QUEUED_AT,
+            interval=timedelta(minutes=15),
+            callback=SyncCallback(
+                custom_synchronous_callback,
+                kwargs={"alert_type": "time_exceeded", "dag_id": "custom_deadline_alert"},
+            ),
+        ),
+    ):
+        EmptyOperator(task_id="example_task")
+
+A **custom asynchronous callback** is only slightly more work.  Note in the following example that
+the custom callback code is placed in a separate file, and must be imported in the DAG file.
+
+.. code-block:: python
+
+    # Place this method in `/files/plugins/deadline_callbacks.py`
+    async def custom_async_callback(**kwargs):
+        """Handle deadline violation with custom logic."""
+        print(f"Deadline exceeded for DAG {kwargs.get("dag_id")}!")
+        print(f"Alert type: {kwargs.get("alert_type")}")
+        # Additional custom handling here
+
+
+    # Place this in a dag file
```

Comment:
```suggestion

.. code-block:: python

    # Place this in a dag file
```

maybe making them 2 blocks?

Reply:
Sure, I'll make the change in the next revision, thanks.